client/lib/common/src/data/application/application-store.js

```diff
@@ -73,6 +93,17 @@ class ApplicationStore extends Store {
     }
 
     /**
+     * Replaces application version with Failed state.
+     *
+     * @param  {error} err The error passed from the flux action.
+     */
+    failFetchApplicationVersion(err) {
+        this.setState({
+            versions: _m.assoc( this.state.versions, err.ver, new Failed( err ) )
```

Comment:
not sure what should be done here.. failFetchApplication is replacing the application id with a Failed result